base/range.jl

```diff
@@ -1380,8 +1380,16 @@ function vcat(rs::AbstractRange{T}...) where T
     return a
 end
 
-Array{T,1}(r::AbstractRange{T}) where {T} = vcat(r)
-collect(r::AbstractRange) = vcat(r)
+function Array{T,1}(r::AbstractRange{T}) where {T}
```

Comment:
Just a minor note from the sidelines: there is a discussion in this PR why this method is needed, and why removing it would be possible, but would require changing code/behavior for which  in turn it is unclear why it is the way it is.

After this PR, there'll be one more layer which when reviewed in a couple years, people will wonder "why is it this way?" and a future PR might also say "it is unclear why it is there" just referencing this method now?

Of course they can dig through old commits, then look up the corresponding PRs, read the discussions there, rinse and repeat ("code archaeology").

Live would be much easier for those poor future developers if there was a brief comment here summarizing this (and if there had been such a comment in the code you were wondering about, I am sure that would have been helpful, too... :-/).

Anyway, just my two cents as a bystander. You do you :-). For what it's worth, I'll appreciate your contribution either way :-)

Reply:
Yes, a comment here would make sense if we decide to retain this method

Reply:
Looking at the git blame, it seems support for non-conforming `UnitRange`s was added intentionally in https://github.com/JuliaLang/julia/pull/27302, and is pre-1.0 behavior(!). In light of this, I'll add a comment now to link to that discussion, but in the future, it might be better to return `StepRange`s automatically in such cases by checking the type of something like `oneadditiveunit`, as discussed on discourse.